### PR TITLE
feat: handle deterministic URLs

### LINKED ISSUE #493: 🐞 [Bug Report] - deterministic urls update
**Describe the bug**

With the release of [Deterministic URLs](https://cloud.google.com/run/docs/release-notes#September_03_2024), Cloud Run services now have two default urls: 

```
SERVICE-PROJECTNUM.REGION.run.app
SERVICE-HASH-REGIONCODE.a.run.app
```

Avocano uses [self-csrf](https://github.com/GoogleCloudPlatform/avocano/blob/main/docs/admin/django-self-csrf.md) to build the hash-regioncode variation of it's own URL, but not the new version. 

This causes issues when trying to use the web-interface for the server, specifically when using the default presented URL in the Google Cloud Console or Service URL in the output of a `gcloud run deploy` (both report the projectnum variation)

**Additional context**

internal b/367711357

### PR DESCRIPTION
**Pull request by glasnt**
## Description

Fixes #493

Changes CLOUDRUN_SERVICE_URL to support a list of URLs from the "run.googleapis.com/urls" annotation for CSRF. 

Opted to not pluralise this URL, but update it to presume it's a string, with possible comma separation. 

This also may not require `run.viewer` on the server service account any more, which will need to be updated with the tf updates. 

## Checklist
- [x] Please **merge** this PR for me once it is approved.

### CODE REVIEW

**Review by NimJay — APPROVED** [review]
Looks good to me.

Thanks for fixing this!

**Comment by NimJay** [comment]
Sorry, I see that you ticked "Please merge this PR for me once it is approved", but I don't have permissions to merge.

**Comment by glasnt** [comment]
I added you to `avocano-approvers` ✨

**Pull request merged**
